src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/tools/JMap.java

```diff
@@ -151,6 +155,18 @@ public static void main(String[] args) {
                                 System.exit(1);
                             }
                             dumpfile = keyValue[1];
+                        } else if (keyValue[0].equals("gz")) {
```

Comment:
I don't see "gz" being rejected if "-heap" or "-heap:format=x" are used. I think it needs to be.